eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java

```diff
@@ -308,9 +309,11 @@ public InstanceInfo(InstanceInfo ii) {
         UNKNOWN;
 
         public static InstanceStatus toEnum(String s) {
-            for (InstanceStatus e : InstanceStatus.values()) {
-                if (e.name().equalsIgnoreCase(s)) {
-                    return e;
+            if (s != null) {
+                try {
+                    return InstanceStatus.valueOf(s.toUpperCase());
+                } catch (IllegalArgumentException e) {
+                    // ignore and fall through to unknown
```

Comment:
Can a debug/trace level message be logged so this defaulting to unknown isn't completely ignored?